funnel/models/project.py

```diff
@@ -592,15 +592,21 @@ def calendar_weeks(self, leading_weeks=True):
                 session_dates.insert(0, (now, None, None, 0))
 
         weeks = defaultdict(dict)
+        today = now.date().isoformat()
         for project_date, day_start_at, day_end_at, session_count in session_dates:
             weekobj = Week.withdate(project_date)
             if weekobj.week not in weeks:
+                # Set upcoming for weeks with event dates in the future
+                weeks[weekobj.week]['upcoming'] = weekobj >= current_week
                 weeks[weekobj.week]['year'] = weekobj.year
                 # Order is important, and we need dict to count easily
                 weeks[weekobj.week]['dates'] = OrderedDict()
             for wdate in weekobj.days():
                 weeks[weekobj.week]['dates'].setdefault(wdate, 0)
                 if project_date.date() == wdate:
+                    # If the event is over don't set upcoming for current week
+                    if wdate.isoformat() < today:
```

Comment:
This is comparing strings. Compare dates directly.

Reply:
Changed to compare dates